lib/trino-filesystem-s3/src/test/java/io/trino/filesystem/s3/AbstractTestS3FileSystem.java

```diff
@@ -65,6 +67,16 @@ final void cleanup()
         fileSystemFactory = null;
     }
 
+    protected GetObjectRequest.Builder configure(GetObjectRequest.Builder requestBuilder)
+    {
+        return requestBuilder;
+    }
+
+    protected PutObjectRequest.Builder configure(PutObjectRequest.Builder requestBuilder)
+    {
+        return requestBuilder;
+    }
```

Comment:
Instead of handing the test - Can we have a `DelegatingS3Client` and we could implement `invokeOperation` accordingly ?

Reply:
Now the changes could be specific to the final test class which is going to run them